frontend/ui/widget/inputdialog.lua

```diff
@@ -723,6 +724,52 @@ function InputDialog:_addScrollButtons(nav_bar)
                 end,
             })
         end
+        -- Add a button to go to the line by its number in the file
+        if self.fullscreen then
+            table.insert(row, {
+                text = "Go",
+                callback = function()
+                    local cur_line_num, last_line_num = self._input_widget:getLineNums()
+                    input_dialog = InputDialog:new{
+                        title = _("Enter line number"),
+                        input_hint = T("%1 (1 - %2)", cur_line_num, last_line_num),
+                        input_type = "number",
+                        stop_events_propagation = true, -- avoid interactions with upper InputDialog
+                        buttons = {
+                            {
+                                {
+                                    text = _("Cancel"),
+                                    callback = function()
+                                        UIManager:close(input_dialog)
+                                        if not self.keyboard_hidden then
+                                            self:onShowKeyboard()
+                                        end
+                                    end,
+                                },
+                                {
+                                    text = _("Go to line"),
+                                    is_enter_default = true,
+                                    callback = function()
+                                        local new_line_num = tonumber(input_dialog:getInputText())
+                                        if new_line_num and new_line_num >= 1 and new_line_num <= last_line_num then
+                                            UIManager:close(input_dialog)
+                                            self._input_widget:onCloseKeyboard()
+                                            self._input_widget:moveCursorToCharPos(self._input_widget:getLineCharPos(new_line_num))
+                                            if not self.keyboard_hidden then
+                                                self:refreshButtons()
+                                                self:onShowKeyboard ()
```

Comment:
Is this needed (closing upper keyboard, to re-open it.
And refreshing buttons only if keyboard is not hidden (the buttons aren't part of the keyboard)?

Reply:
Not needed, and refresh buttons not needed, tested.